src/examples/rpp/doxygen/reduce.cpp

```diff
@@ -0,0 +1,25 @@
+#include <rpp/rpp.hpp>
+
+#include <iostream>
+
+/**
+ * @example reduce.cpp
+ **/
+
+int main()
+{
+    //! [reduce]
+    rpp::source::just(1, 2, 3)
+        | rpp::operators::reduce(10, std::plus<int>{})
```

Comment:
can we have reduce example with different types? like std::string

Reply:
done